src/main/java/frc/robot/subsystems/BatterySimulation.java

```diff
@@ -0,0 +1,65 @@
+package frc.robot.subsystems;
+
+import static org.junit.Assert.assertEquals;
+
+import edu.wpi.first.wpilibj.simulation.RoboRioSim;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.simulation.DriverStationSim;
+import frc.robot.Constants;
+
+public class BatterySimulation extends SubsystemBase {
+  /*
+  Simulate voltage
+  General timer - 2 test case
+      - General Red
+      - General Yellow
+  High current timer - 2 test cases
+      - High Current Red
+      - High Current Yellow
+  */
+  private double voltageToSubtract = 0.0001;
+  private BatterySubsystem batterySubsystem;
+
+  public BatterySimulation(BatterySubsystem batterySubsystem) {
+    this.batterySubsystem = batterySubsystem;
+    RoboRioSim.setVInVoltage(14.0);
+  }
+
+  @Override
+  public void simulationPeriodic() {
```

Comment:
The different tests should be sequenced in a better way. Since the whole method runs each 20 ms loop there are probably some conflicting conditions happening. Using JUnit testing framework may make some of this easier, but using getFPGATimestamp() to only run tests over a specific interval of runtime would also work. Something like this:
```
double now = Timer.getFpgaTimestamp();
if (now >= 0 && now < 1.0) {
    // Do voltage check test
} else if (now >=1.0 and now < 30.0) {
    // Do timer test
}
// else if (etc....)
```

Reply:
fixed, timing is now better